In [84]:
# imports
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [85]:
# load data
boulder = pd.read_csv('data/boulder_results.csv')
lead = pd.read_csv('data/lead_results.csv')
speed = pd.read_csv('data/speed_results.csv')
combined = pd.read_csv('data/combined_results.csv')

# Boulder Category

In [86]:
boulder.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9741 entries, 0 to 9740
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Competition Title  9741 non-null   object 
 1   Competition Date   9741 non-null   object 
 2   FIRST              9741 non-null   object 
 3   LAST               9741 non-null   object 
 4   Nation             9741 non-null   object 
 5   StartNr            9325 non-null   float64
 6   Rank               9741 non-null   int64  
 7   Qualification      5432 non-null   object 
 8   Qualification 1    2154 non-null   object 
 9   Qualification 2    2155 non-null   object 
 10  Semifinal          1524 non-null   object 
 11  Final              1458 non-null   object 
 12  Category           9741 non-null   object 
dtypes: float64(1), int64(1), object(11)
memory usage: 989.4+ KB


In [87]:
boulder.head()

Competition Title         Competition Date  \
0  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
1  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
2  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
3  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
4  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   

    FIRST        LAST Nation  StartNr  Rank Qualification Qualification 1  \
0   Naile     MEIGNAN    FRA     15.0     1        6T7z99             NaN   
1  Miriam        FOGU    ITA     27.0     2       4T8z814             NaN   
2   Vanda  MICHALKOVA    SVK     48.0     3        6T7z89             NaN   
3    Lola     SAUTIER    FRA     17.0     4        4T6z69             NaN   
4  Polina    KULAGINA    RUS     39.0     5       4T6z611             NaN   

  Qualification 2 Semifinal   Final Category  
0             NaN    3T4z55  3T4z89  boulder  
1             NaN    3T4z55  2T4z27  boulder  
2             NaN    3T3z43  2T3z23  boulder  
3             NaN    2T3z89  1T3z38  boulder  
4             NaN    3T4z76  1T3z47  boulder

In [88]:
# checking for duplicated entries
boulder.duplicated().sum()

3272

In [92]:
# remove duplicates
boulder.drop_duplicates(inplace=True)
boulder.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6469 entries, 0 to 6468
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Competition Title  6469 non-null   object 
 1   Competition Date   6469 non-null   object 
 2   FIRST              6469 non-null   object 
 3   LAST               6469 non-null   object 
 4   Nation             6469 non-null   object 
 5   StartNr            6261 non-null   float64
 6   Rank               6469 non-null   int64  
 7   Qualification      2923 non-null   object 
 8   Qualification 1    1774 non-null   object 
 9   Qualification 2    1772 non-null   object 
 10  Semifinal          1162 non-null   object 
 11  Final              859 non-null    object 
 12  Category           6469 non-null   object 
dtypes: float64(1), int64(1), object(11)
memory usage: 707.5+ KB


In [93]:
# extracting the location of the competiton from the Competition Title column into a new column 'Location'
boulder['Location'] = boulder['Competition Title'].str.extract(r'\(([A-Z]{3})\)')
boulder.head()

Competition Title         Competition Date  \
0  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
1  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
2  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
3  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
4  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   

    FIRST        LAST Nation  StartNr  Rank Qualification Qualification 1  \
0   Naile     MEIGNAN    FRA     15.0     1        6T7z99             NaN   
1  Miriam        FOGU    ITA     27.0     2       4T8z814             NaN   
2   Vanda  MICHALKOVA    SVK     48.0     3        6T7z89             NaN   
3    Lola     SAUTIER    FRA     17.0     4        4T6z69             NaN   
4  Polina    KULAGINA    RUS     39.0     5       4T6z611             NaN   

  Qualification 2 Semifinal   Final Category Location  
0             NaN    3T4z55  3T4z89  boulder      ITA  
1             NaN    3T4z55  2T4z27  boulder      ITA  
2             NaN    3T3z43  2T3z23  boulder      ITA  
3             NaN    2T3z89  1T3z38  boulder      ITA  
4             NaN    3T4z76  1T3z47  boulder      ITA

In [94]:
# create a new column 'home' that indicates if the athlete is competeing in their home country

boulder['home'] = boulder['Nation'].eq(boulder['Location'])
boulder.head()

Competition Title         Competition Date  \
0  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
1  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
2  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
3  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   
4  European Youth Championships (B) - Brixen (ITA...  20  - 22 September 2019   

    FIRST        LAST Nation  StartNr  Rank Qualification Qualification 1  \
0   Naile     MEIGNAN    FRA     15.0     1        6T7z99             NaN   
1  Miriam        FOGU    ITA     27.0     2       4T8z814             NaN   
2   Vanda  MICHALKOVA    SVK     48.0     3        6T7z89             NaN   
3    Lola     SAUTIER    FRA     17.0     4        4T6z69             NaN   
4  Polina    KULAGINA    RUS     39.0     5       4T6z611             NaN   

  Qualification 2 Semifinal   Final Category Location   home  
0             NaN    3T4z55  3T4z89  boulder      ITA  False  
1             NaN    3T4z55  2T4z27  boulder      ITA   True  
2             NaN    3T3z43  2T3z23  boulder      ITA  False  
3             NaN    2T3z89  1T3z38  boulder      ITA  False  
4             NaN    3T4z76  1T3z47  boulder      ITA  False

In [95]:
# check to see how many unique values for Nation
boulder_nations = boulder['Nation'].unique()
print(boulder_nations)
print('Climbers are from ', len(boulder_nations), ' different nations')

['FRA' 'ITA' 'SVK' 'RUS' 'AUT' 'ESP' 'SLO' 'BEL' 'BUL' 'SUI' 'NED' 'CRO'
 'UKR' 'NOR' 'CZE' 'GBR' 'GRE' 'DEN' 'SRB' 'LAT' 'IRL' 'ISR' 'POL' 'POR'
 'HUN' 'GER' 'ROU' 'FIN' 'SWE' 'BLR' 'LTU' 'JPN' 'USA' 'CAN' 'KOR' 'RSA'
 'AUS' 'CHI' 'NZL' 'IND' 'MEX' 'BRA' 'UZB' 'ECU' 'KAZ' 'THA' 'INA' 'ARG'
 'HKG' 'SGP' 'LUX' 'CHN' 'IRI' 'TPE' 'PAK' 'PER' 'GUA' 'EST' 'MAC' 'MAS'
 'PHI' 'VEN' 'CAM' 'GEO' 'KGZ' 'MKD' 'LKA' 'TUR' 'MGL' 'NEP']
Climbers are from  70  different nations


In [96]:
# How many unique countries where competitions are held? Also checking for re extraction mistakes
boulder_locations = boulder['Location'].unique()
print(boulder_locations)
print('Competitions are held in', len(boulder_locations), 'different countries')

['ITA' 'POL' 'JPN' 'USA' 'BUL' 'GER' 'AUT' 'CHN' 'POR' 'RUS' 'SUI' 'HKG'
 'ECU' 'BEL' 'THA' 'NED']
Competitions are held in 16 different countries


Only athletes from these 16 countries got chances to compete in their home country.

In [97]:
# How many individual athletes just by first and last name
boulder.groupby(['FIRST','LAST']).size().sort_values(ascending=False)

FIRST            LAST             
Heeyeon          PARK                 16
Flavy            COHAUT               16
Yoshiyuki        OGATA                16
Urska            REPUSIC              16
Alex             KHAZANOV             16
                                      ..
Leonardo         MAGALLANES TEJADA     1
                 GONTERO               1
                 DE RIVERO HUAMAN      1
Lenka            FURDIKOVA             1
ADLIYAH BAIQUNI  EGALITA               1
Length: 1840, dtype: int64

In [98]:
# How many individual athletes by name and nation
boulder.groupby(['FIRST', 'LAST', 'Nation']).size().sort_values(ascending=False)

FIRST            LAST       Nation
Mickael          MAWEM      FRA       16
Urska            REPUSIC    SLO       16
Alex             KHAZANOV   ISR       16
Heeyeon          PARK       KOR       16
Yoshiyuki        OGATA      JPN       16
                                      ..
Hristo           ALTANCHEV  BUL        1
Hongik           CHOI       KOR        1
Rudrangsho       DEY        IND        1
Hinayah          MUHAMMAD   INA        1
ADLIYAH BAIQUNI  EGALITA    INA        1
Length: 1842, dtype: int64

In [99]:
# Athletes with same name but different nation?
w_nation = boulder.groupby(['FIRST', 'LAST', 'Nation']).size().to_frame(name = 'count').reset_index()
no_nation = boulder.groupby(['FIRST', 'LAST']).size().to_frame(name = 'count').reset_index()
difference = pd.concat([w_nation.drop('Nation', 1), no_nation]).drop_duplicates(keep=False)
difference

FIRST    LAST  count
962   Louis  FECHOZ      1
963   Louis  FECHOZ      2
962   Louis  FECHOZ      3
1419  Robin   CASEY      4

In [100]:
boulder.loc[boulder['LAST'] == 'FECHOZ']

Competition Title     Competition Date  \
1830           European Youth Cup (B) - Graz (AUT) 2019    11  - 12 May 2019   
2282          European Youth Cup (B) - Soure (POR) 2019  27  - 28 April 2019   
4394  IFSC Youth World Championships -  Moscow (RUS)...  9  - 16 August 2018   

      FIRST    LAST Nation  StartNr  Rank Qualification Qualification 1  \
1830  Louis  FECHOZ    FRA     74.0     4      8T8z1212             NaN   
2282  Louis  FECHOZ    FRA      NaN     5       7T8z128             NaN   
4394  Louis  FECHOZ    AUS    169.0    33           NaN          2T3z78   

     Qualification 2 Semifinal   Final Category Location   home  
1830             NaN       NaN  3T3z88  boulder      AUT  False  
2282             NaN       NaN  2T3z35  boulder      POR  False  
4394             NaN       NaN     NaN  boulder      RUS  False

According to the IFSC website
Louis Fechoz is Australian but he is on the French team.

In [101]:
boulder.loc[boulder['LAST'] == 'CASEY']

Competition Title  \
44    European Youth Championships (B) - Brixen (ITA...   
423   IFSC Youth World Championships -  Arco (ITA) 2...   
3651  European Youth Championships (B) - Brussels (B...   
4554         European Youth Cup (B) - Sofia (BUL) 2018    

                  Competition Date  FIRST   LAST Nation  StartNr  Rank  \
44         20  - 22 September 2019  Robin  CASEY    IRL     23.0    45   
423           22  - 31 August 2019  Robin  CASEY    IRL    370.0    51   
3651  31 August - 2 September 2018  Robin  CASEY    GBR    603.0    26   
4554            21  - 22 July 2018  Robin  CASEY    GBR     70.0    16   

     Qualification Qualification 1 Qualification 2 Semifinal Final Category  \
44          1T2z43             NaN             NaN       NaN   NaN  boulder   
423            NaN             NaN          0T1z02       NaN   NaN  boulder   
3651       2T5z210             NaN             NaN       NaN   NaN  boulder   
4554        4T4z54             NaN             NaN       NaN   NaN  boulder   

     Location   home  
44        ITA  False  
423       ITA  False  
3651      BEL  False  
4554      BUL  False


Robin Casey was on the British team until she switch to team Ireland in 2019.

In [116]:
boulder[['FIRST', 'LAST', 'home', 'Rank']].sort_values(by=['FIRST', 'LAST', 'home'], ignore_index=True).head(50)

FIRST         LAST   home  Rank
0   ADLIYAH BAIQUNI      EGALITA  False    77
1          ALEXANDR       PETROV  False    18
2             ALINA  PONOMARYOVA  False    77
3             ALINA  PONOMARYOVA  False    29
4             ALINA  PONOMARYOVA  False    63
5            ANDREY  BURZHINSKIY  False    81
6           ARAILYM     ONGALBAY  False    21
7           ARAILYM     ONGALBAY  False    56
8             Aaron       MATTES  False    79
9             Aaron    PEÑARANDA  False    71
10             Abby       GEBERT  False    51
11             Abby       GEBERT  False    51
12             Abby      MANNING  False    51
13         Abhishek        MEHTA  False    38
14             Adam    ADAMOVSKY  False     6
15             Adam    ADAMOVSKY  False     4
16             Adam    ADAMOVSKY  False    12
17             Adam        ONDRA  False     6
18             Adam        ONDRA  False     5
19             Adam        ONDRA  False     2
20             Adam        ONDRA  False    14
21             Adam        ONDRA  False     2
22             Adam        ONDRA  False     1
23             Adam        ONDRA  False    17
24            Adela     BAJEROVA  False    41
25            Adela     BAJEROVA  False    36
26            Adela     BAJEROVA  False    29
27         Adelaide    D'ADDARIO  False    19
28         Adelaide    D'ADDARIO  False     5
29         Adelaide    D'ADDARIO  False    17
30         Adelaide    D'ADDARIO  False     2
31         Adelaide    D'ADDARIO  False     6
32         Adelaide    D'ADDARIO  False     9
33         Adelaide    D'ADDARIO  False    13
34         Adelaide    D'ADDARIO   True    39
35         Adelaide    D'ADDARIO   True    23
36          Adeliya     UTESHEVA  False    73
37          Adeliya     UTESHEVA  False    27
38          Adeliya     UTESHEVA  False    51
39         Adil Bin       AZMOON  False    28
40           Adrian      CHMIALA   True    28
41           Adrien      LEMAIRE  False    25
42           Afonso       MENDES   True    47
43      Afonso Mano     CRISANTO  False    47
44      Afonso Mano     CRISANTO  False    65
45      Afonso Mano     CRISANTO  False    59
46      Afonso Mano     CRISANTO  False    36
47      Afonso Mano     CRISANTO  False    26
48      Afonso Mano     CRISANTO  False    28
49      Afonso Mano     CRISANTO   True    46

In [146]:
# Create dataframe of rank averages when they compete at home or away for each athlete
ra_boulder = boulder[['FIRST', 'LAST', 'home', 'Rank']].groupby(['FIRST', 'LAST', 'home']).mean().reset_index()
ra_boulder

FIRST           LAST   home        Rank
0     ADLIYAH BAIQUNI        EGALITA  False   77.000000
1            ALEXANDR         PETROV  False   18.000000
2               ALINA    PONOMARYOVA  False   56.333333
3              ANDREY    BURZHINSKIY  False   81.000000
4             ARAILYM       ONGALBAY  False   38.500000
...               ...            ...    ...         ...
2167             Ádám       BAKURECZ  False   26.000000
2168            Örjan  RÖDLAND VAAGE  False   38.666667
2169            Övgün       YILDIRIM  False  125.000000
2170            Šimon        POTŮČEK  False   34.000000
2171           Štěpán        POTŮČEK  False   37.714286

[2172 rows x 4 columns]

In [170]:
# Checking for duplicated athletes
ra_boulder.duplicated(subset=['FIRST', 'LAST']).sum()

332

332 athletes have competed at home and away.

In [147]:
# Just checking random athletes for fun
ra_boulder.loc[rank_average['LAST'] == 'PILZ']

FIRST  LAST   home       Rank
888  Jessica  PILZ  False  12.636364
889  Jessica  PILZ   True   4.000000

In [148]:
ra_boulder.loc[rank_average['LAST'] == 'HARADA']

FIRST    LAST   home  Rank
984   Kai  HARADA  False  14.5
985   Kai  HARADA   True   7.5

In [149]:
ra_boulder.loc[rank_average['LAST'] == 'GARNBRET']

FIRST      LAST   home      Rank
862  Janja  GARNBRET  False  1.090909

In [150]:
ra_boulder.loc[rank_average['LAST'] == 'ONDRA']

FIRST   LAST   home      Rank
11  Adam  ONDRA  False  6.714286

In [151]:
ra_boulder.loc[rank_average['LAST'] == 'BAILEY']

FIRST    LAST   home       Rank
1771  Sean  BAILEY  False  39.090909
1772  Sean  BAILEY   True  10.000000

In [152]:
ra_boulder.loc[rank_average['LAST'] == 'NARASAKI']

FIRST      LAST   home       Rank
1340  Meichi  NARASAKI  False  19.500000
1341  Meichi  NARASAKI   True   8.000000
1958   Tomoa  NARASAKI  False   4.363636
1959   Tomoa  NARASAKI   True   1.500000

In [153]:
ra_boulder.loc[rank_average['LAST'] == 'CONDIE']

FIRST    LAST   home       Rank
1058  Kyra  CONDIE  False  18.166667
1059  Kyra  CONDIE   True   9.500000

In [102]:
# unique competitons

print('There are ', len(boulder['Competition Title'].unique()), 'bouldering competitions in 2018-19')

boulder.groupby('Competition Title').size()

There are  33 bouldering competitions in 2018-19


Competition Title
Asia Cup (B) - Hong Kong (HKG) 2018                                   73
Asia Cup (B,S) - Bangkok (THA) 2018                                   59
Asian Championships - Kurayoshi (JPN) 2018                            87
Asian Cup (B) - Hong Kong (HKG) 2019                                  49
Asian Youth Championships - Chongqing (CHN) 2018                     151
European Championship (B) - Zakopane (POL) 2019                       91
European Youth Championships (B) - Brixen (ITA) 2019                 282
European Youth Championships (B) - Brussels (BEL) 2018               301
European Youth Cup (B) - Delft (NED) 2018                            211
European Youth Cup (B) - Graz (AUT) 2018                             275
European Youth Cup (B) - Graz (AUT) 2019                             328
European Youth Cup (B) - Sofia (BUL) 2018                            193
European Youth Cup (B) - Sofia (BUL) 2019                            258
European Youth Cup (B) - Soure (P

# Lead Category

In [156]:
lead.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9930 entries, 0 to 9929
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Competition Title  9930 non-null   object 
 1   Competition Date   9930 non-null   object 
 2   FIRST              9930 non-null   object 
 3   LAST               9930 non-null   object 
 4   Nation             9930 non-null   object 
 5   StartNr            9480 non-null   float64
 6   Rank               9930 non-null   int64  
 7   Qualification      648 non-null    object 
 8   Qualification 1    9034 non-null   object 
 9   Qualification 2    9694 non-null   object 
 10  Semifinal          2446 non-null   object 
 11  Final              1716 non-null   object 
 12  Points             8737 non-null   float64
 13  Category           9930 non-null   object 
dtypes: float64(2), int64(1), object(11)
memory usage: 1.1+ MB


In [157]:
lead.head()

Competition Title     Competition Date  \
0  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
1  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
2  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
3  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
4  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   

     FIRST      LAST Nation  StartNr  Rank Qualification  \
0    Lucka   RAKOVEC    SLO    222.0     1           NaN   
1    Laura    ROGORA    ITA    234.0     2           NaN   
2     Luce    DOUADY    FRA    229.0     3           NaN   
3     Mina  MARKOVIC    SLO    237.0     4           NaN   
4  Jessica      PILZ    AUT    213.0     5           NaN   

               Qualification 1              Qualification 2 Semifinal Final  \
0      50+1.,50+1.,50+1.,50+1.      44 2.,44 2.,44 2.,44 2.        39   46+   
1  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       39+   45+   
2      48+3.,48+3.,48+3.,48+3.  30+15.,30+15.,30+15.,30+15.       38+    43   
3  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       43+   42+   
4      46+5.,46+5.,46+5.,46+5.      42+3.,42+3.,42+3.,42+3.       38+    42   

   Points Category  
0    1.73     lead  
1   15.62     lead  
2    7.83     lead  
3   15.62     lead  
4    3.87     lead

In [158]:
# Checking for duplicates
lead.duplicated().sum()

0

In [159]:
# extracting the location of the competiton from the Competition Title column into a new column 'Location'
lead['Location'] = lead['Competition Title'].str.extract(r'\(([A-Z]{3})\)')
lead.head()

Competition Title     Competition Date  \
0  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
1  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
2  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
3  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
4  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   

     FIRST      LAST Nation  StartNr  Rank Qualification  \
0    Lucka   RAKOVEC    SLO    222.0     1           NaN   
1    Laura    ROGORA    ITA    234.0     2           NaN   
2     Luce    DOUADY    FRA    229.0     3           NaN   
3     Mina  MARKOVIC    SLO    237.0     4           NaN   
4  Jessica      PILZ    AUT    213.0     5           NaN   

               Qualification 1              Qualification 2 Semifinal Final  \
0      50+1.,50+1.,50+1.,50+1.      44 2.,44 2.,44 2.,44 2.        39   46+   
1  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       39+   45+   
2      48+3.,48+3.,48+3.,48+3.  30+15.,30+15.,30+15.,30+15.       38+    43   
3  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       43+   42+   
4      46+5.,46+5.,46+5.,46+5.      42+3.,42+3.,42+3.,42+3.       38+    42   

   Points Category Location  
0    1.73     lead      GBR  
1   15.62     lead      GBR  
2    7.83     lead      GBR  
3   15.62     lead      GBR  
4    3.87     lead      GBR

In [160]:
# create a new column 'home' that indicates if the athlete is competeing in their home country

lead['home'] = lead['Nation'].eq(lead['Location'])
lead.head()

Competition Title     Competition Date  \
0  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
1  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
2  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
3  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   
4  European Championship (L,S) - Edinburgh (GBR) ...  4  - 6 October 2019   

     FIRST      LAST Nation  StartNr  Rank Qualification  \
0    Lucka   RAKOVEC    SLO    222.0     1           NaN   
1    Laura    ROGORA    ITA    234.0     2           NaN   
2     Luce    DOUADY    FRA    229.0     3           NaN   
3     Mina  MARKOVIC    SLO    237.0     4           NaN   
4  Jessica      PILZ    AUT    213.0     5           NaN   

               Qualification 1              Qualification 2 Semifinal Final  \
0      50+1.,50+1.,50+1.,50+1.      44 2.,44 2.,44 2.,44 2.        39   46+   
1  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       39+   45+   
2      48+3.,48+3.,48+3.,48+3.  30+15.,30+15.,30+15.,30+15.       38+    43   
3  16+24.,16+24.,16+24.,16+24.      36+7.,36+7.,36+7.,36+7.       43+   42+   
4      46+5.,46+5.,46+5.,46+5.      42+3.,42+3.,42+3.,42+3.       38+    42   

   Points Category Location   home  
0    1.73     lead      GBR  False  
1   15.62     lead      GBR  False  
2    7.83     lead      GBR  False  
3   15.62     lead      GBR  False  
4    3.87     lead      GBR  False

In [161]:
# check to see how many unique values for Nation
lead_nations = lead['Nation'].unique()
print(lead_nations)
print('Climbers are from ', len(lead_nations), ' different nations')

['SLO' 'ITA' 'FRA' 'AUT' 'NOR' 'BEL' 'SUI' 'GBR' 'UKR' 'SVK' 'NED' 'RUS'
 'ISR' 'POL' 'CZE' 'ESP' 'GER' 'CRO' 'GRE' 'KOR' 'JPN' 'USA' 'IRI' 'TPE'
 'HKG' 'BRA' 'CAN' 'SWE' 'ARG' 'CHI' 'AUS' 'IRL' 'RSA' 'DEN' 'BUL' 'MEX'
 'NZL' 'UZB' 'ECU' 'LUX' 'HUN' 'LAT' 'IND' 'THA' 'SGP' 'ROU' 'POR' 'KGZ'
 'CHN' 'KAZ' 'INA' 'GEO' 'VEN' 'PER' 'GUA' 'PHI' 'CAM' 'MAC' 'MAS' 'SRB'
 'EST' 'BLR' 'MGL' 'MKD' 'TUR' 'FIN']
Climbers are from  66  different nations


In [162]:
# How many unique countries where competitions are held? Also checking for re extraction mistakes
lead_locations = lead['Location'].unique()
print(lead_locations)
print('Competitions are held in', len(lead_locations), 'different countries')

['GBR' 'SLO' 'ITA' 'JPN' 'AUT' 'FRA' 'SUI' 'ECU' 'CHN' 'RUS' 'GER' nan]
Competitions are held in 12 different countries


In [207]:
# investigate nan value in Location
lead['Competition Title'].unique()

array(['European Championship (L,S) - Edinburgh (GBR) 2019',
       'IFSC Climbing Worldcup (L) - Kranj (SLO) 2019 ',
       'IFSC Youth World Championships -  Arco (ITA) 2019 ',
       'IFSC Climbing World Championships - Hachioji (JPN) 2019',
       'European Youth Cup (S,L) - Imst (AUT) 2019',
       'IFSC Climbing Worldcup (L) - Briançon (FRA) 2019 ',
       'IFSC Climbing Worldcup (L, S) - Chamonix (FRA) 2019 ',
       'IFSC Climbing Worldcup (L, S) - Villars (SUI) 2019',
       'European Youth Cup (L) - St. Pierre Faucigny (FRA) 2019',
       'European Youth Cup (L) - Ostermundigen (SUI) 2019',
       'IFSC PanAmerican Championship (L, S, B, C) - Guayaquil (ECU) 2018',
       'Asian Championships - Kurayoshi (JPN) 2018 ',
       'Asian Youth Championships - Chongqing (CHN) 2018',
       'IFSC Climbing Worldcup (L,S) - Xiamen (CHN) 2018 ',
       'IFSC Climbing Worldcup (L,S) - Wujiang (CHN) 2018 ',
       'IFSC Climbing Worldcup (L) - Kranj (SLO) 2018 ',
       'IFSC Climbing Wor

The Imst 2018 competition does not have a country ISO code. It should be AUT 


In [164]:
# fill the nan values with 'AUT'
lead['Location'].fillna('AUT', inplace=True)
lead['Location'].unique()

array(['GBR', 'SLO', 'ITA', 'JPN', 'AUT', 'FRA', 'SUI', 'ECU', 'CHN',
       'RUS', 'GER'], dtype=object)

In [165]:
# how many individual athletes by names only?
lead.groupby(['FIRST','LAST']).size().sort_values(ascending=False)

FIRST        LAST             
Mykhayil     TKACHUK              37
Laura        ROGORA               33
Alberto      GINÉS LÓPEZ          33
Jakub        KONECNY              32
Mikel Asier  LINACISORO MOLINA    32
                                  ..
Georgii      KEDROV                2
Winai        RUANGRIT              2
Nikola       KRAMARIC              1
Veronika     SCHEUEROVA            1
Sergei       LUZHETSKII            1
Length: 1467, dtype: int64

In [166]:
# how many individul athletes by names and nations?
lead.groupby(['FIRST', 'LAST', 'Nation']).size().sort_values(ascending=False)

FIRST        LAST                 Nation
Mykhayil     TKACHUK              UKR       37
Laura        ROGORA               ITA       33
Alberto      GINÉS LÓPEZ          ESP       33
Jakub        KONECNY              CZE       32
Mikel Asier  LINACISORO MOLINA    ESP       32
                                            ..
Mariia       MUSIENKO             RUS        2
Frederik     VIBERG CHRISTIANSEN  DEN        2
Nikola       KRAMARIC             CRO        1
Veronika     SCHEUEROVA           CZE        1
Sergei       LUZHETSKII           RUS        1
Length: 1468, dtype: int64

In [167]:
# Athletes with same name but different nation?
w_nation2 = lead.groupby(['FIRST', 'LAST', 'Nation']).size().to_frame(name = 'count').reset_index()
no_nation2 = lead.groupby(['FIRST', 'LAST']).size().to_frame(name = 'count').reset_index()
difference2 = pd.concat([w_nation2.drop('Nation', 1), no_nation2]).drop_duplicates(keep=False)
difference2

FIRST   LAST  count
1119  Robin  CASEY      8

In [168]:
lead.loc[lead['LAST'] == 'CASEY']

Competition Title  \
249   IFSC Youth World Championships -  Arco (ITA) 2...   
871          European Youth Cup (S,L) - Imst (AUT) 2019   
3868        European Youth Cup (L) - Munich (GER) 2018    
4250         European Youth Cup (L) - Uster (SUI) 2018    
5173  IFSC Youth World Championships -  Arco (ITA) 2...   
5795         European Youth Cup (S,L) - Imst (AUT) 2019   
8792        European Youth Cup (L) - Munich (GER) 2018    
9174         European Youth Cup (L) - Uster (SUI) 2018    

           Competition Date  FIRST   LAST Nation  StartNr  Rank Qualification  \
249    22  - 31 August 2019  Robin  CASEY    IRL    370.0    43           NaN   
871      2  - 4 August 2019  Robin  CASEY    IRL    150.0    40           NaN   
3868       7  - 8 July 2018  Robin  CASEY    GBR     16.0    23           NaN   
4250  30 June - 1 July 2018  Robin  CASEY    GBR     30.0    31           NaN   
5173   22  - 31 August 2019  Robin  CASEY    IRL    370.0    43           NaN   
5795     2  - 4 August 2019  Robin  CASEY    IRL    150.0    40           NaN   
8792       7  - 8 July 2018  Robin  CASEY    GBR     16.0    23           NaN   
9174  30 June - 1 July 2018  Robin  CASEY    GBR     30.0    31           NaN   

                                        Qualification 1  \
249                         30+30.,30+30.,30+30.,30+30.   
871                         28+42.,28+42.,28+42.,28+42.   
3868                        24+26.,24+26.,24+26.,24+26.   
4250                        19 33.,19 33.,19 33.,19 33.   
5173  30+30.,30+30.,30+30.,30+30.,30+30.,30+30.,30+3...   
5795  28+42.,28+42.,28+42.,28+42.,28+42.,28+42.,28+4...   
8792  24+26.,24+26.,24+26.,24+26.,24+26.,24+26.,24+2...   
9174  19 33.,19 33.,19 33.,19 33.,19 33.,19 33.,19 3...   

                                        Qualification 2 Semifinal Final  \
249                         12+58.,12+58.,12+58.,12+58.       NaN   NaN   
871                         27+36.,27+36.,27+36.,27+36.       NaN   NaN   
3868                        25+15.,25+15.,25+15.,25+15.       NaN   NaN   
4250                        25+27.,25+27.,25+27.,25+27.       NaN   NaN   
5173  12+58.,12+58.,12+58.,12+58.,12+58.,12+58.,12+5...       NaN   NaN   
5795  27+36.,27+36.,27+36.,27+36.,27+36.,27+36.,27+3...       NaN   NaN   
8792  25+15.,25+15.,25+15.,25+15.,25+15.,25+15.,25+1...       NaN   NaN   
9174  25+27.,25+27.,25+27.,25+27.,25+27.,25+27.,25+2...       NaN   NaN   

      Points Category Location   home  
249    43.45     lead      ITA  False  
871    39.15     lead      AUT  False  
3868   20.91     lead      GER  False  
4250   31.58     lead      SUI  False  
5173   43.45     lead      ITA  False  
5795   39.15     lead      AUT  False  
8792   20.91     lead      GER  False  
9174   31.58     lead      SUI  False

This is the same athlete that switched team from GBR to IRL whom we encounter before. But there seems to be duplicated entries with scorings in different format so it was not caught by the method before. 

In [179]:
# checking for duplicates again using subset of columns.
lead.duplicated(subset=['Competition Title', 'Competition Date', 'FIRST', 'LAST', 'Nation', 'Rank']).sum()

4924

In [182]:
# drop duplicated
lead.drop_duplicates(subset=['Competition Title', 'Competition Date', 'FIRST', 'LAST', 'Nation', 'Rank'], inplace=True)
lead.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5006 entries, 0 to 5005
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Competition Title  5006 non-null   object 
 1   Competition Date   5006 non-null   object 
 2   FIRST              5006 non-null   object 
 3   LAST               5006 non-null   object 
 4   Nation             5006 non-null   object 
 5   StartNr            4781 non-null   float64
 6   Rank               5006 non-null   int64  
 7   Qualification      324 non-null    object 
 8   Qualification 1    4561 non-null   object 
 9   Qualification 2    4888 non-null   object 
 10  Semifinal          1249 non-null   object 
 11  Final              866 non-null    object 
 12  Points             4781 non-null   float64
 13  Category           5006 non-null   object 
 14  Location           5006 non-null   object 
 15  home               5006 non-null   bool   
dtypes: bool(1), float64(2), 

In [183]:
# Create dataframe of rank averages when they compete at home or away for each athlete
ra_lead = lead[['FIRST', 'LAST', 'home', 'Rank']].groupby(['FIRST', 'LAST', 'home']).mean().reset_index()
ra_lead

FIRST                LAST   home       Rank
0     ALEXANDR              PETROV  False  41.000000
1        ALINA         PONOMARYOVA  False  40.500000
2      ARAILYM            ONGALBAY  False  40.500000
3        Aaron           PEÑARANDA  False  51.000000
4        Aaron           PEÑARANDA   True   4.000000
...        ...                 ...    ...        ...
1739      Àlex  HERNÁNDEZ CASTILLA  False  73.500000
1740     Óscar       MONTÓN GARCÍA  False  58.000000
1741     Örjan       RÖDLAND VAAGE  False  37.800000
1742     Šimon             POTŮČEK  False  35.307692
1743    Štěpán             POTŮČEK  False  33.600000

[1744 rows x 4 columns]

In [184]:
# Checking for duplicated athletes
ra_lead.duplicated(subset=['FIRST', 'LAST']).sum()

277

277 athletes have competed both at home and away.

In [186]:
# checking random athletes
ra_lead.loc[ra_lead['LAST'] == 'GARNBRET']

FIRST      LAST   home      Rank
666  Janja  GARNBRET  False  2.090909
667  Janja  GARNBRET   True  7.500000

In [190]:
ra_lead.loc[ra_lead['LAST'] == 'PILZ']

FIRST  LAST   home  Rank
693  Jessica  PILZ  False   3.5
694  Jessica  PILZ   True   1.0

In [193]:
ra_lead.loc[ra_lead['LAST'] == 'NOGUCHI']

FIRST     LAST   home      Rank
31  Akiyo  NOGUCHI  False  5.857143
32  Akiyo  NOGUCHI   True  3.500000

In [194]:
ra_lead.loc[ra_lead['LAST'] == 'ROGORA']

FIRST    LAST   home       Rank
854  Laura  ROGORA  False  17.071429
855  Laura  ROGORA   True   7.666667

In [195]:
ra_lead.loc[ra_lead['LAST'] == 'MORI']

FIRST  LAST   home  Rank
21    Ai  MORI  False   3.5
22    Ai  MORI   True   3.0

In [196]:
ra_lead.loc[ra_lead['LAST'] == 'GHISOLFI']

FIRST      LAST   home       Rank
270   Claudia  GHISOLFI  False  37.750000
271   Claudia  GHISOLFI   True  28.000000
1514  Stefano  GHISOLFI  False   8.666667
1515  Stefano  GHISOLFI   True   2.000000

# Speed Category

In [197]:
speed.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11868 entries, 0 to 11867
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Competition Title  11868 non-null  object
 1   Competition Date   11868 non-null  object
 2   FIRST              11868 non-null  object
 3   LAST               11868 non-null  object
 4   Nation             11868 non-null  object
 5   StartNr            11868 non-null  int64 
 6   Rank               11868 non-null  int64 
 7   Qualification      11868 non-null  object
 8   1/8 - Final        2656 non-null   object
 9   1/4 - Final        2290 non-null   object
 10  1/2 - Final        1181 non-null   object
 11  Small final        592 non-null    object
 12  Final              589 non-null    object
 13  Category           11868 non-null  object
dtypes: int64(2), object(12)
memory usage: 1.3+ MB


In [198]:
speed.head()

Competition Title       Competition Date  \
0  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
1  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
2  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
3  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
4  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   

      FIRST      LAST Nation  StartNr  Rank Qualification 1/8 - Final  \
0      Anna   CALANCA    ITA      114     1         8.686         NaN   
1  Capucine  VIGLIONE    FRA       67     2         8.612         NaN   
2     Lison   GAUTRON    FRA       66     3         8.977         NaN   
3    Polina  KULAGINA    RUS      159     4         8.872         NaN   
4   Kamilla  KUSHAEVA    RUS      160     5         8.871         NaN   

  1/4 - Final 1/2 - Final Small final   Final Category  
0       8.855       9.442         NaN   8.661    speed  
1       9.243       8.605         NaN  10.175    speed  
2       8.546      13.409       8.886     NaN    speed  
3       8.837       9.249       9.302     NaN    speed  
4       8.639         NaN         NaN     NaN    speed

In [199]:
# Checking for duplicated entries
speed.duplicated().sum()

7843

In [200]:
# Dropping duplicates
speed.drop_duplicates(inplace=True)
speed.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4025 entries, 0 to 4024
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Competition Title  4025 non-null   object
 1   Competition Date   4025 non-null   object
 2   FIRST              4025 non-null   object
 3   LAST               4025 non-null   object
 4   Nation             4025 non-null   object
 5   StartNr            4025 non-null   int64 
 6   Rank               4025 non-null   int64 
 7   Qualification      4025 non-null   object
 8   1/8 - Final        896 non-null    object
 9   1/4 - Final        790 non-null    object
 10  1/2 - Final        407 non-null    object
 11  Small final        204 non-null    object
 12  Final              203 non-null    object
 13  Category           4025 non-null   object
dtypes: int64(2), object(12)
memory usage: 471.7+ KB


In [201]:
# extracting the location of the competiton from the Competition Title column into a new column 'Location'
speed['Location'] = speed['Competition Title'].str.extract(r'\(([A-Z]{3})\)')
speed.head()

Competition Title       Competition Date  \
0  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
1  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
2  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
3  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
4  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   

      FIRST      LAST Nation  StartNr  Rank Qualification 1/8 - Final  \
0      Anna   CALANCA    ITA      114     1         8.686         NaN   
1  Capucine  VIGLIONE    FRA       67     2         8.612         NaN   
2     Lison   GAUTRON    FRA       66     3         8.977         NaN   
3    Polina  KULAGINA    RUS      159     4         8.872         NaN   
4   Kamilla  KUSHAEVA    RUS      160     5         8.871         NaN   

  1/4 - Final 1/2 - Final Small final   Final Category Location  
0       8.855       9.442         NaN   8.661    speed      RUS  
1       9.243       8.605         NaN  10.175    speed      RUS  
2       8.546      13.409       8.886     NaN    speed      RUS  
3       8.837       9.249       9.302     NaN    speed      RUS  
4       8.639         NaN         NaN     NaN    speed      RUS

In [202]:
speed['home'] = speed['Nation'].eq(speed['Location'])
speed.head()

Competition Title       Competition Date  \
0  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
1  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
2  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
3  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   
4  European Youth Championships (L,S) - Voronezh ...  18  - 20 Oktober 2019   

      FIRST      LAST Nation  StartNr  Rank Qualification 1/8 - Final  \
0      Anna   CALANCA    ITA      114     1         8.686         NaN   
1  Capucine  VIGLIONE    FRA       67     2         8.612         NaN   
2     Lison   GAUTRON    FRA       66     3         8.977         NaN   
3    Polina  KULAGINA    RUS      159     4         8.872         NaN   
4   Kamilla  KUSHAEVA    RUS      160     5         8.871         NaN   

  1/4 - Final 1/2 - Final Small final   Final Category Location   home  
0       8.855       9.442         NaN   8.661    speed      RUS  False  
1       9.243       8.605         NaN  10.175    speed      RUS  False  
2       8.546      13.409       8.886     NaN    speed      RUS  False  
3       8.837       9.249       9.302     NaN    speed      RUS   True  
4       8.639         NaN         NaN     NaN    speed      RUS   True

In [203]:
# check to see how many unique values for Nation
speed_nations = speed['Nation'].unique()
print(speed_nations)
print('Climbers are from ', len(speed_nations), ' different nations')

['ITA' 'FRA' 'RUS' 'GER' 'BUL' 'AUT' 'SLO' 'CRO' 'NOR' 'ESP' 'SVK' 'DEN'
 'CZE' 'UKR' 'POL' 'HUN' 'GBR' 'ISR' 'SUI' 'GRE' 'USA' 'INA' 'KOR' 'KAZ'
 'BEL' 'CAN' 'AUS' 'JPN' 'NZL' 'RSA' 'IND' 'IRL' 'BRA' 'LAT' 'SWE' 'CHI'
 'THA' 'MEX' 'HKG' 'ECU' 'SGP' 'UZB' 'CHN' 'TPE' 'IRI' 'ARG' 'NED' 'PAK'
 'GEO' 'MAS' 'BLR' 'VEN' 'PER' 'GUA' 'PHI' 'CAM' 'KGZ' 'SRB' 'LKA' 'MGL'
 'MKD' 'TUR']
Climbers are from  62  different nations


In [204]:
# How many unique countries where competitions are held? Also checking for re extraction mistakes
speed_locations = speed['Location'].unique()
print(speed_locations)
print('Competitions are held in', len(speed_locations), 'different countries')

['RUS' 'GBR' 'ITA' 'JPN' 'AUT' 'POL' 'FRA' 'SUI' 'CHN' 'ECU' 'THA' nan]
Competitions are held in 12 different countries


In [206]:
# investigate nan value 
speed['Competition Title'].unique()

array(['European Youth Championships (L,S) - Voronezh (RUS) 2019',
       'European Championship (L,S) - Edinburgh (GBR) 2019',
       'IFSC Youth World Championships -  Arco (ITA) 2019 ',
       'IFSC Climbing World Championships - Hachioji (JPN) 2019',
       'European Youth Cup (S,L) - Imst (AUT) 2019',
       'European Youth Cup (S) - Tarnow (POL) 2019',
       'IFSC Climbing Worldcup (L, S) - Chamonix (FRA) 2019 ',
       'IFSC Climbing Worldcup (L, S) - Villars (SUI) 2019',
       'European Youth Cup (S) - Mezzolombardo (ITA) 2019',
       'IFSC Climbing Worldcup (B,S) - Wujiang (CHN) 2019',
       'IFSC Climbing Worldcup (B,S) - Chongqing (CHN) 2019',
       'IFSC Climbing Worldcup (B,S) - Moscow (RUS) 2019 ',
       'IFSC PanAmerican Championship (L, S, B, C) - Guayaquil (ECU) 2018',
       'Asian Championships - Kurayoshi (JPN) 2018 ',
       'Asian Youth Championships - Chongqing (CHN) 2018',
       'IFSC Climbing Worldcup (L,S) - Xiamen (CHN) 2018 ',
       'IFSC Climbing Wo

Same problem as before: The ISO code missing from the competition title from Imst 2018

In [208]:
# fill the nan values with 'AUT'
speed['Location'].fillna('AUT', inplace=True)
speed['Location'].unique()

array(['RUS', 'GBR', 'ITA', 'JPN', 'AUT', 'POL', 'FRA', 'SUI', 'CHN',
       'ECU', 'THA'], dtype=object)

In [209]:
# how many individual athletes by names only?
speed.groupby(['FIRST','LAST']).size().sort_values(ascending=False)

FIRST            LAST             
Gian Luca        ZODDA                19
Elena            REMIZOVA             17
Hana             KRIZOVA              17
Aleksandra       KALUCKA              17
Marcin           DZIENSKI             16
                                      ..
Leonid           OSADCHYI              1
Leonardo         VAZQUEZ RODRIGUEZ     1
Lela             HENTSCHEL             1
Leila            SHMIDKE               1
ADLIYAH BAIQUNI  EGALITA               1
Length: 1191, dtype: int64

In [210]:
# how many individual athletes by name and nation?
speed.groupby(['FIRST', 'LAST', 'Nation']).size().sort_values(ascending=False)

FIRST            LAST               Nation
Gian Luca        ZODDA              ITA       19
Elena            REMIZOVA           RUS       17
Aleksandra       KALUCKA            POL       17
Hana             KRIZOVA            CZE       17
Aurelia          SARISSON           FRA       16
                                              ..
Leonid           OSADCHYI           UKR        1
Leonardo         VAZQUEZ RODRIGUEZ  MEX        1
Lela             HENTSCHEL          SUI        1
Leila            SHMIDKE            KAZ        1
ADLIYAH BAIQUNI  EGALITA            INA        1
Length: 1192, dtype: int64

In [211]:
# Athletes with same name but different nation?
w_nation3 = speed.groupby(['FIRST', 'LAST', 'Nation']).size().to_frame(name = 'count').reset_index()
no_nation3 = speed.groupby(['FIRST', 'LAST']).size().to_frame(name = 'count').reset_index()
difference3 = pd.concat([w_nation3.drop('Nation', 1), no_nation3]).drop_duplicates(keep=False)
difference3

FIRST    LAST  count
632  Louis  FECHOZ      1
633  Louis  FECHOZ      3
632  Louis  FECHOZ      4

This is the same athlete we encountered before who is a Australian national but competes for the French team.

In [212]:
speed.loc[speed['LAST'] == 'FECHOZ']

Competition Title  \
65    European Youth Championships (L,S) - Voronezh ...   
755          European Youth Cup (S,L) - Imst (AUT) 2019   
878          European Youth Cup (S) - Tarnow (POL) 2019   
2686  IFSC Youth World Championships -  Moscow (RUS)...   

           Competition Date  FIRST    LAST Nation  StartNr  Rank  \
65    18  - 20 Oktober 2019  Louis  FECHOZ    FRA       83    14   
755      2  - 4 August 2019  Louis  FECHOZ    FRA       61     3   
878      19  - 20 July 2019  Louis  FECHOZ    FRA       80     5   
2686    9  - 16 August 2018  Louis  FECHOZ    AUS      298    26   

     Qualification 1/8 - Final 1/4 - Final 1/2 - Final Small final Final  \
65           9.312         NaN         NaN         NaN         NaN   NaN   
755          8.340         NaN       8.020       8.390       7.970   NaN   
878          7.720         NaN       7.690         NaN         NaN   NaN   
2686         9.330         NaN         NaN         NaN         NaN   NaN   

     Category Location   home  
65      speed      RUS  False  
755     speed      AUT  False  
878     speed      POL  False  
2686    speed      RUS  False

In [213]:
# Create dataframe of rank averages when they compete at home or away for each athlete
ra_speed = speed[['FIRST', 'LAST', 'home', 'Rank']].groupby(['FIRST', 'LAST', 'home']).mean().reset_index()
ra_speed

FIRST                LAST   home       Rank
0     ADLIYAH BAIQUNI             EGALITA  False  11.000000
1            ALEXANDR              PETROV  False  22.000000
2               ALINA         PONOMARYOVA  False  21.333333
3              ANDREY         BURZHINSKIY  False  41.000000
4             ARAILYM            ONGALBAY  False  28.500000
...               ...                 ...    ...        ...
1420             pino                LEON  False  38.000000
1421           solene              MOREAU  False  11.000000
1422        valentine              MANGIN  False   7.000000
1423             Àlex  HERNÁNDEZ CASTILLA  False  71.000000
1424            Örjan       RÖDLAND VAAGE  False  28.000000

[1425 rows x 4 columns]

In [240]:
ra_speed.duplicated(subset=['FIRST', 'LAST']).sum()

234

234 athletes have both home and away entries.

In [227]:
# skip for now
sa_speed = speed[['FIRST', 'LAST', 'home', 'Qualification', '1/8 - Final', '1/4 - Final', '1/2 - Final', 'Small final', 'Final']]
sa_speed

FIRST              LAST   home Qualification 1/8 - Final 1/4 - Final  \
0         Anna           CALANCA  False         8.686         NaN       8.855   
1     Capucine          VIGLIONE  False         8.612         NaN       9.243   
2        Lison           GAUTRON  False         8.977         NaN       8.546   
3       Polina          KULAGINA   True         8.872         NaN       8.837   
4      Kamilla          KUSHAEVA   True         8.871         NaN       8.639   
...        ...               ...    ...           ...         ...         ...   
4020  Mykhayil           TKACHUK  False         9.780         NaN         NaN   
4021   Guillem     MONSECH GASCA  False        10.230         NaN         NaN   
4022   Kristóf              TÓTH  False        10.270         NaN         NaN   
4023     Jorge  DÍAZ-RULLO CALVO  False        11.910         NaN         NaN   
4024    Mathis          BROSSIER   True        12.600         NaN         NaN   

     1/2 - Final Small final   Final  
0          9.442         NaN   8.661  
1          8.605         NaN  10.175  
2         13.409       8.886     NaN  
3          9.249       9.302     NaN  
4            NaN         NaN     NaN  
...          ...         ...     ...  
4020         NaN         NaN     NaN  
4021         NaN         NaN     NaN  
4022         NaN         NaN     NaN  
4023         NaN         NaN     NaN  
4024         NaN         NaN     NaN  

[4025 rows x 9 columns]

# Combined Category

In [69]:
combined.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 879 entries, 0 to 878
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Competition Title      879 non-null    object 
 1   Competition Date       879 non-null    object 
 2   FIRST                  879 non-null    object 
 3   LAST                   879 non-null    object 
 4   Nation                 879 non-null    object 
 5   Rank                   879 non-null    int64  
 6   Qualification lead     879 non-null    object 
 7   Qualification speed    879 non-null    object 
 8   Qualification boulder  879 non-null    object 
 9   Final lead             120 non-null    object 
 10  Final speed            120 non-null    object 
 11  Final boulder          120 non-null    object 
 12  Points                 879 non-null    float64
 13  Final Points           120 non-null    float64
 14  Category               879 non-null    object 
dtypes: flo

In [70]:
combined.head()

Competition Title      Competition Date  \
0  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
1  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
2  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
3  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
4  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   

     FIRST        LAST Nation  Rank Qualification lead Qualification speed  \
0  Natsumi      HIRANO    JPN     1                 1.                 14.   
1    Julia        LOTZ    AUT     2                 7.                  1.   
2    Emily    PHILLIPS    GBR     3                 2.                  5.   
3     Jana       RAUTH    AUT     4                 4.                  4.   
4    Vanda  MICHALKOVA    SVK     5                 3.                 15.   

  Qualification boulder Final lead Final speed Final boulder  Points  \
0                    1.        NaN         NaN           NaN    14.0   
1                    5.        NaN         NaN           NaN    35.0   
2                    4.        NaN         NaN           NaN    40.0   
3                    3.        NaN         NaN           NaN    48.0   
4                    6.        NaN         NaN           NaN   270.0   

   Final Points  Category  
0           NaN  combined  
1           NaN  combined  
2           NaN  combined  
3           NaN  combined  
4           NaN  combined

In [228]:
# checking for duplicates
combined.duplicated().sum()

586

In [230]:
# dropping duplicates
combined.drop_duplicates(inplace=True)
combined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 293 entries, 0 to 292
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Competition Title      293 non-null    object 
 1   Competition Date       293 non-null    object 
 2   FIRST                  293 non-null    object 
 3   LAST                   293 non-null    object 
 4   Nation                 293 non-null    object 
 5   Rank                   293 non-null    int64  
 6   Qualification lead     293 non-null    object 
 7   Qualification speed    293 non-null    object 
 8   Qualification boulder  293 non-null    object 
 9   Final lead             40 non-null     object 
 10  Final speed            40 non-null     object 
 11  Final boulder          40 non-null     object 
 12  Points                 293 non-null    float64
 13  Final Points           40 non-null     float64
 14  Category               293 non-null    object 
dtypes: flo

In [231]:
# how many combined competition were there?b
combined.groupby('Competition Title').size()

Competition Title
IFSC Climbing World Championships Combined - Hachioji (JPN) 2019      40
IFSC Climbing World Championships Combined - Innsbruck (AUT) 2018     12
IFSC PanAmerican Championship (L, S, B, C) - Guayaquil (ECU) 2018     41
IFSC Youth World Championships -  Arco (ITA) 2019                    200
dtype: int64

In [232]:
# extracting the location of the competiton from the Competition Title column into a new column 'Location'
combined['Location'] = combined['Competition Title'].str.extract(r'\(([A-Z]{3})\)')
combined.head()

Competition Title      Competition Date  \
0  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
1  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
2  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
3  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
4  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   

     FIRST        LAST Nation  Rank Qualification lead Qualification speed  \
0  Natsumi      HIRANO    JPN     1                 1.                 14.   
1    Julia        LOTZ    AUT     2                 7.                  1.   
2    Emily    PHILLIPS    GBR     3                 2.                  5.   
3     Jana       RAUTH    AUT     4                 4.                  4.   
4    Vanda  MICHALKOVA    SVK     5                 3.                 15.   

  Qualification boulder Final lead Final speed Final boulder  Points  \
0                    1.        NaN         NaN           NaN    14.0   
1                    5.        NaN         NaN           NaN    35.0   
2                    4.        NaN         NaN           NaN    40.0   
3                    3.        NaN         NaN           NaN    48.0   
4                    6.        NaN         NaN           NaN   270.0   

   Final Points  Category Location  
0           NaN  combined      ITA  
1           NaN  combined      ITA  
2           NaN  combined      ITA  
3           NaN  combined      ITA  
4           NaN  combined      ITA

In [233]:
combined['home'] = combined['Nation'].eq(combined['Location'])
combined.head()

Competition Title      Competition Date  \
0  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
1  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
2  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
3  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   
4  IFSC Youth World Championships -  Arco (ITA) 2...  22  - 31 August 2019   

     FIRST        LAST Nation  Rank Qualification lead Qualification speed  \
0  Natsumi      HIRANO    JPN     1                 1.                 14.   
1    Julia        LOTZ    AUT     2                 7.                  1.   
2    Emily    PHILLIPS    GBR     3                 2.                  5.   
3     Jana       RAUTH    AUT     4                 4.                  4.   
4    Vanda  MICHALKOVA    SVK     5                 3.                 15.   

  Qualification boulder Final lead Final speed Final boulder  Points  \
0                    1.        NaN         NaN           NaN    14.0   
1                    5.        NaN         NaN           NaN    35.0   
2                    4.        NaN         NaN           NaN    40.0   
3                    3.        NaN         NaN           NaN    48.0   
4                    6.        NaN         NaN           NaN   270.0   

   Final Points  Category Location   home  
0           NaN  combined      ITA  False  
1           NaN  combined      ITA  False  
2           NaN  combined      ITA  False  
3           NaN  combined      ITA  False  
4           NaN  combined      ITA  False

In [234]:
# check to see how many unique values for Nation
combined_nations = combined['Nation'].unique()
print(combined_nations)
print('Climbers are from ', len(combined_nations), ' different nations')

['JPN' 'AUT' 'GBR' 'SVK' 'BEL' 'SUI' 'BUL' 'SLO' 'KOR' 'ESP' 'CRO' 'POL'
 'NOR' 'AUS' 'GRE' 'CHI' 'NZL' 'IRL' 'DEN' 'HUN' 'IND' 'SWE' 'BRA' 'LAT'
 'ITA' 'CZE' 'CAN' 'THA' 'MEX' 'USA' 'GER' 'ISR' 'UKR' 'HKG' 'SGP' 'UZB'
 'ECU' 'RSA' 'FRA' 'CHN' 'KAZ' 'ARG' 'VEN' 'PER' 'GUA']
Climbers are from  45  different nations


In [235]:
# How many unique countries where competitions are held? Also checking for re extraction mistakes
combined_locations = combined['Location'].unique()
print(combined_locations)
print('Competitions are held in', len(combined_locations), 'different countries')

['ITA' 'JPN' 'ECU' 'AUT']
Competitions are held in 4 different countries


In [236]:
# how many individual athletes by names only?
combined.groupby(['FIRST','LAST']).size().sort_values(ascending=False)

FIRST    LAST       
Petra    KLINGLER       2
Rudolph  RUANA          2
Tomoa    NARASAKI       2
Kokoro   FUJII          2
Miho     NONAKA         2
                       ..
Matthew  JONES          1
Mattea   PÖTZI          1
Mathias  RASK JUNKER    1
Mateus   BELLOTTO       1
Abby     GEBERT         1
Length: 277, dtype: int64

In [237]:
# how many individual athletes by name and nation?
combined.groupby(['FIRST', 'LAST', 'Nation']).size().sort_values(ascending=False)

FIRST    LAST         Nation
Petra    KLINGLER     SUI       2
Rudolph  RUANA        USA       2
Tomoa    NARASAKI     JPN       2
Kokoro   FUJII        JPN       2
Miho     NONAKA       JPN       2
                               ..
Matthew  JONES        NZL       1
Mattea   PÖTZI        AUT       1
Mathias  RASK JUNKER  DEN       1
Mateus   BELLOTTO     BRA       1
Abby     GEBERT       NZL       1
Length: 277, dtype: int64

In [238]:
# Create dataframe of rank averages when they compete at home or away for each athlete
ra_combined = combined[['FIRST', 'LAST', 'home', 'Rank']].groupby(['FIRST', 'LAST', 'home']).mean().reset_index()
ra_combined

FIRST         LAST   home  Rank
0          Abby       GEBERT  False  30.0
1          Abby      MANNING  False  21.0
2          Adam        ONDRA  False  10.0
3            Ai         MORI   True   6.0
4         Akiyo      NOGUCHI  False   4.0
..          ...          ...    ...   ...
280         Zoi  PALTATSIDOU  False  35.0
281         Zoé         EGLI  False   8.0
282     Zuzanna      MIENTUS  False  29.0
283  jose tomas      LEDESMA  False  30.0
284        leah     JEFFRIES  False  22.0

[285 rows x 4 columns]

In [239]:
ra_combined.duplicated(subset=['FIRST', 'LAST']).sum()

8

8 athletes have home and away entries.

In [243]:
ra_combined.loc[ra_combined['LAST'] == 'NOGUCHI']

FIRST     LAST   home  Rank
4  Akiyo  NOGUCHI  False   4.0
5  Akiyo  NOGUCHI   True   2.0